packages/components/src/InputGroup/InputGroup.module.css

```diff
@@ -30,22 +30,27 @@
   --public-field--bottom-right-radius: var(--radius-base);
 }
 
-/* First Horizontal Input Group */
-
-.horizontal:first-child > :first-child {
+/* Horizontal: restore outer rounding on left/right items */
+.horizontal > :first-child:not(.horizontal) {
   --public-field--top-left-radius: var(--radius-base);
+  --public-field--bottom-left-radius: var(--radius-base);
 }
 
-.horizontal:first-child > :last-child {
+.horizontal > :last-child:not(.horizontal) {
   --public-field--top-right-radius: var(--radius-base);
+  --public-field--bottom-right-radius: var(--radius-base);
 }
 
-/* Last Horizontal Input Group */
-
-.horizontal:last-child > :first-child {
-  --public-field--bottom-left-radius: var(--radius-base);
+/* Nested horizontal groups inside a horizontal parent */
+/* If the nested group is the first child, round its first field's left corners */
+/* If the nested group is not at the left edge, flatten its first field's left corners */
+.horizontal > .horizontal:not(:first-child) > :first-child {
+  --public-field--top-left-radius: 0;
+  --public-field--bottom-left-radius: 0;
 }
 
-.horizontal:last-child > :last-child {
-  --public-field--bottom-right-radius: var(--radius-base);
+/* If the nested group is not at the right edge, flatten its last field's right corners */
+.horizontal > .horizontal:not(:last-child) > :last-child {
```

Comment:
I think we're going to need more rules. If you look in the Nested story - a vertical group containing horizontal groups)  it seems like the css is giving rounded corners to the inner fields of the horizontal groups.

<img width="667" height="230" alt="Screenshot 2025-09-17 at 8 43 31 PM" src="https://github.com/user-attachments/assets/38c427e3-8408-40c1-b383-9eccc2cb5803" />

I think we need...
- if a horizontal group is the first child in a vertical group, give first field rounded top left corner/if horizontal group is last child give last field rounded bottom right corner
  - and if horizontal group is in the middle of a vertical group, its fields get no rounded corners